include/exadg/postprocessor/pressure_difference_calculation.cpp

```diff
@@ -48,7 +47,20 @@ PressureDifferenceCalculator<dim, Number>::setup(DoFHandler<dim> const & dof_han
   data                 = data_in;
 
   if(data.calculate)
+  {
     create_directories(data.directory, mpi_comm);
+    evaluation_cache = std::make_shared<Utilities::MPI::RemotePointEvaluation<dim>>(
+      1e-8 * dof_handler_pressure->begin()->diameter());
+
+    // only request result on rank 0
+    if(Utilities::MPI::this_mpi_process(mpi_comm) == 0)
+    {
+      std::vector<Point<dim>> points{data.point_1, data.point_2};
+      evaluation_cache->reinit(points, dof_handler_pressure->get_triangulation(), mapping_in);
+    }
+    else
+      evaluation_cache->reinit({}, dof_handler_pressure->get_triangulation(), mapping_in);
+  }
```

Comment:
Why not filling the vector ``points`` in the if-else and then call the function ``reinit()`` only once afterwards? The same comment applies to other parts of the code with the same structure.